supertokens/userIdMapping.go

```diff
@@ -0,0 +1,217 @@
+	}
+	if maxVersion(cdiVersion, "2.15") != cdiVersion {
+		return CreateUserIdMappingResult{}, errors.New("Please upgrade the SuperTokens core to >= 3.15.0")
+	}
+
+	data := map[string]interface{}{
+		"superTokensUserId": supertokensUserId,
+		"externalUserId":    externalUserId,
+		"force":             force,
+	}
+	if externalUserIdInfo != nil {
+		data["externalUserIdInfo"] = *externalUserIdInfo
+	}
+	resp, err := querier.SendPostRequest("/recipe/userid/map", data)
+	if err != nil {
+		return CreateUserIdMappingResult{}, err
+	}
+	if resp["status"] == "OK" {
+		return CreateUserIdMappingResult{
+			OK: &struct{}{},
+		}, nil
+	} else if resp["status"] == "UNKNOWN_SUPERTOKENS_USER_ID_ERROR" {
+		return CreateUserIdMappingResult{
+			UnknownSupertokensUserIdError: &struct{}{},
+		}, nil
+	} else {
+		return CreateUserIdMappingResult{
+			UserIdMappingAlreadyExistsError: &struct {
+				DoesSuperTokensUserIdExist bool
+				DoesExternalUserIdExist    bool
+			}{
+				DoesSuperTokensUserIdExist: resp["doesSuperTokensUserIdExist"].(bool),
+				DoesExternalUserIdExist:    resp["doesExternalUserIdExist"].(bool),
+			},
+		}, nil
+	}
+}
+
+type GetUserIdMappingResult struct {
+	OK *struct {
+		SupertokensUserId  string
+		ExternalUserId     string
+		ExternalUserIdInfo *string
+	}
+	UnknownMappingError *struct{}
+}
+
+func GetUserIdMapping(userId string, userIdType *string) (GetUserIdMappingResult, error) {
+	if userIdType == nil {
+		userIdType = &userIdTypeDefault
+	}
+	err := validateUserIdType(*userIdType)
+	if err != nil {
+		return GetUserIdMappingResult{}, err
+	}
+
+	querier, err := GetNewQuerierInstanceOrThrowError("")
+	if err != nil {
+		return GetUserIdMappingResult{}, err
+	}
+	cdiVersion, err := querier.GetQuerierAPIVersion()
+	if err != nil {
+		return GetUserIdMappingResult{}, err
+	}
+	if maxVersion(cdiVersion, "2.15") != cdiVersion {
+		return GetUserIdMappingResult{}, errors.New("Please upgrade the SuperTokens core to >= 3.15.0")
+	}
+
+	resp, err := querier.SendGetRequest("/recipe/userid/map", map[string]string{
+		"userId":     userId,
+		"userIdType": *userIdType,
+	})
+	if err != nil {
+		return GetUserIdMappingResult{}, err
+	}
+	if resp["status"] == "OK" {
+		var externalUserIdInfo *string = nil
+		if v, ok := resp["externalUserIdInfo"].(string); ok {
+			externalUserIdInfo = &v
+		}
+		return GetUserIdMappingResult{
+			OK: &struct {
+				SupertokensUserId  string
+				ExternalUserId     string
+				ExternalUserIdInfo *string
+			}{
+				SupertokensUserId:  resp["superTokensUserId"].(string),
+				ExternalUserId:     resp["externalUserId"].(string),
+				ExternalUserIdInfo: externalUserIdInfo,
+			},
+		}, nil
+	} else {
+		return GetUserIdMappingResult{
+			UnknownMappingError: &struct{}{},
+		}, nil
+	}
+}
+
+type DeleteUserIdMappingResult struct {
+	OK *struct {
+		DidMappingExist bool
+	}
+}
+
+func DeleteUserIdMapping(userId string, userIdType *string, force bool) (DeleteUserIdMappingResult, error) {
```

Comment:
force bool should be optional with default value as false.

Reply:
Done